src/interface/viewport.c

```diff
@@ -1105,6 +1105,44 @@ bool sub_98197C(
 
 /**
  *
+ *  rct2: 0x006D6258
+ */
+static void vehicle_visual_observation_tower(int x, int imageDirection, int y, int z, rct_vehicle *vehicle, int rct2VehiclePtrFormat)
```

Comment:
You dont need to pass the vehicle ptr format variable you could just pass the `vehicleEntry`. I've missed that in vehicle_visual_roto_drop as well could you change it?